src/ethz_snow/snowing.py

```diff
@@ -0,0 +1,2227 @@
+
+"""Implementation of the SNOWing class.
+
+This module contains the SNOWing class used to run the single vial simulation
+with spatial information.
+"""
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import multiprocessing as mp
+import matplotlib.pyplot as plt
+from scipy.integrate import simps
+from typing import Optional
+
+import ethz_snow.utils as Utils
+from ethz_snow.constants import calculateDerived
+from ethz_snow.operatingConditions import OperatingConditions
+
+HEATFLOW_REQUIREDKEYS = ("int", "ext", "s0")
+VIAL_GROUPS = ("corner", "edge", "center", "all")
+
+
+class Snowing:
+    """A class to handle a single SNOWing (Stochastic Nucleation of Water
+     with INternal Gradients) simulation.
+
+    More information regarding the equations and their derivation can be found in
+    XXX, Deck et al. (2023).
+
+    Parameters:
+        k (dict): Heat transfer coefficients.
+        opcond (OperatingConditions): Operating conditions of the run.
+        temperature (str): Model dimensionality.
+        configuration (str): Freezing configuration.
+        plotting (bool): Whether evolutions are plotted or not.
+        Nrep (int): Number of repetitions.
+        configPath (Optional[str]): The path of the (optional) custom config yaml.
+    """
+
+    def __init__(
+        self,
+        k: dict = {"int": 0, "ext": 0, "s0": 50, "s_sigma_rel": 0},
+        opcond: OperatingConditions = OperatingConditions(),
+        temperature: str = "spatial_1D",
+        configuration: str = "shelf",
+        plotting: bool = True,
+        Nrep: int = 1,
+        configPath: Optional[str] = None,
+    ):
+        """Construct a Snowing object.
+
+        Args:
+            k (dict, optional): Heat transfer coefficients. Defaults to {"int": 0, "ext": 0, "s0": 50, "s_sigma_rel": 0}.
+            opcond (OperatingConditions, optional): Operating conditions of the run. Defaults to OperatingConditions().
+            temperature (str, optional): Model dimensionality. Defaults to None.
+            configuration (str, optional): Freezing configuration. Defaults to None.
+            plotting (bool, optional): Whether evolutions are plotted or not. Defaults to None.
+            Nrep (int, optional): Number of repetitions. Defaults to 1.
+            configPath (Optional[str], optional): The path of the (optional) custom config yaml. Defaults to None.
+
+        Raises:
+            TypeError: If k is not a dict.
+            ValueError: If a aspecific key is missing in dict k.
+            NotImplementedError: If freezing configuration is not correctly specified.
+            NotImplementedError: If model dimensionality is not correctly specified.
+            TypeError: If opcond is not of type operatingConditions.
+
+        Examples:
+            >>> S = Snowing()
+            >>> S = Snowing(k=d, opcond=op)
+            >>> S = Snowing(k=d, opcond=op, temperature="spatial_1D", configuration="shelf", plotting=True)
```

Comment:
This is not a standalone example, what are `d` and `op`?

Reply:
Should I just remove the input parameters k and opcond here, like in the snowflake examples?